crates/frui_widgets/src/container.rs

```diff
@@ -64,10 +68,52 @@ impl<W: Widget> RenderWidget for Container<W> {
     fn paint(&self, ctx: &mut PaintCtx<Self>, canvas: &mut Canvas, offset: &Offset) {
         if let Some(color) = &self.color {
             let brush = &canvas.solid_brush(color.clone());
-
-            RenderContext::fill(canvas, Rect::from_origin_size(offset, ctx.size()), brush);
+            canvas.fill(DruidRect::from_origin_size(offset, ctx.size()), brush);
         }
 
         ctx.child(0).paint(canvas, offset)
     }
 }
+
+#[derive(RenderWidget, Builder)]
+pub struct DecoratedBox<W: Widget, D: Decoration> {
+    pub child: W,
+    pub decoration: D,
+    pub position: DecorationPosition,
+}
+
+impl DecoratedBox<(), DefaultBoxDecoration> {
+    pub fn builder() -> Self {
+        Self {
+            child: (),
+            decoration: BoxDecoration::builder(),
+            position: DecorationPosition::Background,
+        }
+    }
+}
+
+impl<W: Widget, D: Decoration> RenderWidget for DecoratedBox<W, D> {
+    fn build<'w>(&'w self, _ctx: BuildCtx<'w, Self>) -> Vec<Self::Widget<'w>> {
+        vec![&self.child]
+    }
+
+    fn layout(&self, ctx: &LayoutCtx<Self>, constraints: Constraints) -> Size {
+        self.decoration.padding().inflate_size(ctx.child(0).layout(constraints))
```

Comment:
Here `padding` means size actually. maybe size or dimension is a proper name. Sometimes decoration has its own size

Reply:
The problem is with constraints. `Size` returned from the `layout` must "fit" inside incoming `constraints` (you would normally call `constraints.constrain(size)`). But because we're `inflating` size returned by child (and that child can take full available space) resulting `Size` is bigger than what `constraints` allow.

That's why I get the following warning:
```
[WARN] `DecoratedBox` overflowed
```

Reply:
I wanted to check this so I skimmed through Flutter, but I didn't find a place where they `inflate` this size of decoration

Reply:
> [WARN] `DecoratedBox` overflowed

I didn't mention this, but there was some bug with `Flex` implementation (it was overflowing when it shouldn't). I pushed a fix to this branch.

Reply:
nice catch, I was bothered by this but failed to find it

Reply:
> I wanted to check this so I skimmed through Flutter, but I didn't find a place where they `inflate` this size of decoration

I may be confused with the container part of this; I will check and correct it.